### PR TITLE
Entity properties

### PR DESCRIPTION
**Pull request by tal5**
## Additions
`EntityTag.invulnerable_duration` - paper only, controls the invulnerability duration of a wither
`EntityTag.arms_raised` - paper only, controls whether a ranged entity's arms are raised
`EntityTag.goat_ram` - paper only, causes a goat to ram the specified entity
`EntityTag.material` controls the material enderman are holding / minecarts are carrying, replaces enderman support in `EntityTag.item` (may need a deprecation added in DenizenCore)

### CODE REVIEW

File: paper/src/main/java/com/denizenscript/denizen/paper/properties/EntityArmsRaised.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        // Sets whether a ranged entity is "charging" up an attack (it's arms are raised).
+        // Some entities may require <@link mechanism EntityTag.is_aware> to be set to false.
+        // @tags
+        // <EntityTag.arms_raised>
+        // -->
+        if (mechanism.matches("arms_raised") && mechanism.requireBoolean()) {
+            getRanged().setChargingAttack(mechanism.getValue().asBoolean());
+        }
+    }
+}
```
Comment (mcmonkey4eva) [inline paper/src/main/java/com/denizenscript/denizen/paper/properties/EntityArmsRaised.java]: NL@EoF

File: paper/src/main/java/com/denizenscript/denizen/paper/properties/EntityArmsRaised.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    public static void registerTags() {
+
+        // <--[tag]
+        // @attribute <EntityTag.arms_raised>
+        // @returns ElementTag(Boolean)
+        // @mechanism EntityTag.arms_raised
+        // @group properties
+        // @Plugin Paper
+        // @description
+        // Returns whether a ranged entity is "charging" up an attack (it's arms are raised).
```
Comment (mcmonkey4eva) [inline paper/src/main/java/com/denizenscript/denizen/paper/properties/EntityArmsRaised.java]: The meta should clarify what 'a ranged entity' is

File: paper/src/main/java/com/denizenscript/denizen/paper/properties/EntityGoatRam.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+    EntityTag entity;
+
+    public static void registerTags() {
+        // None
+    }
+
+    @Override
+    public String getPropertyString() {
+        return null;
```
Comment (mcmonkey4eva) [inline paper/src/main/java/com/denizenscript/denizen/paper/properties/EntityGoatRam.java]: This isn't a property, it shouldn't be separated from the general paper entity extensions

**Comment by tal5** [comment]
Implemented changes from review

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/PropertyRegistry.java [hunk 4]
```diff
@@ -106,6 +106,7 @@ public static void registerMainProperties() {
         PropertyParser.registerProperty(EntityJumpStrength.class, EntityTag.class);
         PropertyParser.registerProperty(EntityKnockback.class, EntityTag.class);
         PropertyParser.registerProperty(EntityMarker.class, EntityTag.class);
+        PropertyParser.registerProperty(EntityMaterial.class, EntityTag.class);
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/PropertyRegistry.java:109]: Was discussed on discord iirc - this has nothing to do with the other parts of this same PR

**Pull request closed without merging**